### PR TITLE
feat: endorsement chain provider handling

### PR DESCRIPTION
**Pull request by osslgtm**
## Summary

Endorsement Chain Error/Timeout Handling

## Changes

Endorsement Chain
ChainInfo - for Mumbai RPC

### CODE REVIEW

File: src/constants/chain-info.ts [hunk 1]
```diff
@@ -103,7 +103,7 @@ export const ChainInfo: ChainInfo = {
     networkName: "maticmum",
     networkLabel: "Polygon Mumbai",
     explorerUrl: "https://mumbai.polygonscan.com",
-    rpcUrl: "https://rpc-mumbai.maticvigil.com",
+    rpcUrl: `https://polygon-mumbai.infura.io/v3/${INFURA_API_KEY}`,
     nativeCurrency: {
       name: "MATIC",
       symbol: "mMATIC",
```
Comment (isaackps) [inline src/constants/chain-info.ts:109]: does infura work? 
Also, need to update Polygon mainnet with infura RPC url.

File: src/common/hooks/useEndorsementChain/useEndorsementChain.ts [hunk 2]
```diff
@@ -36,11 +36,8 @@ export const useEndorsementChain = (
       const transferEvents = mergeTransfers([...titleEscrowLogs, ...tokenLogs]);
       const retrievedEndorsementChain = await getEndorsementChain(provider, transferEvents);
       setEndorsementChain(retrievedEndorsementChain);
-    } catch (e) {
-      if (e instanceof Error) {
-        console.error(e);
-        setError(e.message);
-      }
+    } catch (e: any) {
+      setError(e?.message);
```
Comment (isaackps) [inline src/common/hooks/useEndorsementChain/useEndorsementChain.ts]: is there an error that was missed out of instanceof Error? 
can take a look at this article: https://kentcdodds.com/blog/get-a-catch-block-error-message-with-typescript

best is to dont use `any` in typescript

Don't need over kill like in this post, but I think revert back can Liao.

Comment (osslgtm) [reply src/common/hooks/useEndorsementChain/useEndorsementChain.ts]: yeap, ethers is throwing out an object that's not `instanceof Error` but has e.message

File: src/constants/chain-info.ts (hunk 1 above)
Comment (osslgtm) [reply src/constants/chain-info.ts:109]: updated accordingly

File: src/common/hooks/useEndorsementChain/useEndorsementChain.ts [hunk 3]
```diff
@@ -51,3 +48,30 @@ export const useEndorsementChain = (
...
+  try {
+    return new Error(JSON.stringify(maybeError));
+  } catch {
+    return new Error(String(maybeError));
+  }
+};
+
+export const getErrorMessage = (error: unknown): string => {
+  return toErrorWithMessage(error).message;
+};
```
Comment (isaackps) [inline src/common/hooks/useEndorsementChain/useEndorsementChain.ts]: Should this be in a common place so other places that handles error can use it too.

**Review by isaackps — APPROVED** [review]
LGTM!

**Pull request merged**